service_response.go

```diff
@@ -41,33 +42,40 @@ func (e AuthenticationError) Error() string {
 
 // AuthenticationResponse captures authenticated user information
 type AuthenticationResponse struct {
-	User                string         // Users login name
-	ProxyGrantingTicket string         // Proxy Granting Ticket
-	Proxies             []string       // List of proxies
-	AuthenticationDate  time.Time      // Time at which authentication was performed
-	IsNewLogin          bool           // Whether new authentication was used to grant the service ticket
-	IsRememberedLogin   bool           // Whether a long term token was used to grant the service ticket
-	MemberOf            []string       // List of groups which the user is a member of
-	Attributes          UserAttributes // Additional information about the user
+	User                string          // Users login name
+	ProxyGrantingTicket string          // Proxy Granting Ticket
+	Proxies             []string        // List of proxies
+	AuthenticationDate  time.Time       // Time at which authentication was performed
+	IsNewLogin          bool            // Whether new authentication was used to grant the service ticket
+	IsRememberedLogin   bool            // Whether a long term token was used to grant the service ticket
+	MemberOf            []string        // List of groups which the user is a member of
+	Attributes          *UserAttributes // Additional information about the user
 }
 
 // UserAttributes represents additional data about the user
-type UserAttributes map[string][]string
+type UserAttributes struct {
```

Comment:
Changing this breaks the API contract which allows a user to access multiple values of a key. Similar to the `net/http.Header` type.

Reply:
However, the usage scenarios are different. I might use it as a session store.